index.js

```diff
@@ -27,7 +39,70 @@ app.use(express.json());
 // IDEAS METHODS
 
 app.get('/ideas', (req, res) => {
-    res.status(200).json(getIdeas());
+    console.log(req.query);
+    try {
+        if (req.query.sortBy === 'totalDonationSum') {
+            res.status(200).json(sortedByDonationSumIdeas());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'accepted') {
+            res.status(200).json(approvedIdeasList());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'pending') {
+            res.status(200).json(pendingIdeasList());
+            return;
+        }
+        if (req.query.sortBy === 'status' && req.query.status === 'rejected') {
+            res.status(200).json(rejectedIdeasList());
+            return;
+        }
+        res.status(200).json(getIdeas());
+    } catch (error) {
+        res.status(400).send(error.message);
+    }
+});
+
+app.post('/ideas', (req, res) => {
+    try {
+        createIdeas(
+            req.body.header,
+            req.body.description,
+            req.body.askedSum,
+            req.body.userId
+        );
+    } catch (error) {
+        res.status(400).send(error.message);
+    }
+    res.status(200).send('Idea created successfully.');
+});
+
+app.put('/ideas', (req, res) => {
+    try {
+        updateIdea(req.body.ideaId, req.body.key, req.body.value);
+    } catch (error) {
+        res.status(404).send(error.message);
+    }
+    res.status(200).send('Idea updated successfully.');
+});
+
+app.put('/ideas/status', (req, res) => {
+    try {
+        ideasStatusApproval(req.body.ideaId, req.body.isApproved);
```

Comment:
You have a function called `updateIdea` - should this function name follow the same naming pattern?

Reply:
Yes, I think you are right. The hardest thing is to think of the name for variables\function. Fixed. Commiting it.